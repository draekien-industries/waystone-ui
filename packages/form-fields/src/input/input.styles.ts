import { alpha } from '@theme-ui/color';
import { outlineCss, cssSelectors } from '@waystone/css-presets';
import type { FullWidthAttributes, WidthAttributes } from '@waystone/types';
import type { ThemeUICSSObject } from 'theme-ui';

export type InputCssOptions = WidthAttributes & FullWidthAttributes;

export const inputCss = ({
  width,
  minWidth,
  maxWidth,
  fullWidth,
}: InputCssOptions): ThemeUICSSObject => ({
  borderRadius: 'small',
  border: '1px solid',
  borderColor: 'b-400',
  padding: 'extra-small',
  variant: 'text.body',
  color: 'text',
  width: fullWidth ? '100%' : width,
  minWidth,
  maxWidth,
  backgroundColor: alpha('ghost', 0.5),
  ...outlineCss.base,
  [cssSelectors.focusVisible]: {
    ...outlineCss.focused,
  },
  [cssSelectors.active]: {
    ...outlineCss.active,
  },
});
