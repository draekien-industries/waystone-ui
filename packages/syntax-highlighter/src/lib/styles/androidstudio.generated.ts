export const androidstudio = {
  're code.hljs': {
    'display': 'block',
    'overflowX': 'auto',
    'padding': '1em'
  },
  'ode.hljs': {
    'padding': '3px 5px'
  },
  'hljs': {
    'color': '#a9b7c6',
    'background': '#282b2e'
  },
  'hljs-bullet': {
    'color': '#6897bb'
  },
  'hljs-literal': {
    'color': '#6897bb'
  },
  'hljs-number': {
    'color': '#6897bb'
  },
  'hljs-symbol': {
    'color': '#6897bb'
  },
  'hljs-deletion': {
    'color': '#cc7832'
  },
  'hljs-keyword': {
    'color': '#cc7832'
  },
  'hljs-selector-tag': {
    'color': '#cc7832'
  },
  'hljs-link': {
    'color': '#629755'
  },
  'hljs-template-variable': {
    'color': '#629755'
  },
  'hljs-variable': {
    'color': '#629755'
  },
  'hljs-comment': {
    'color': 'grey'
  },
  'hljs-quote': {
    'color': 'grey'
  },
  'hljs-meta': {
    'color': '#bbb529'
  },
  'hljs-addition': {
    'color': '#6a8759'
  },
  'hljs-attribute': {
    'color': '#6a8759'
  },
  'hljs-string': {
    'color': '#6a8759'
  },
  'hljs-section': {
    'color': '#ffc66d'
  },
  'hljs-title': {
    'color': '#ffc66d'
  },
  'hljs-type': {
    'color': '#ffc66d'
  },
  'hljs-name': {
    'color': '#e8bf6a'
  },
  'hljs-selector-class': {
    'color': '#e8bf6a'
  },
  'hljs-selector-id': {
    'color': '#e8bf6a'
  },
  'hljs-emphasis': {
    'fontStyle': 'italic'
  },
  'hljs-strong': {
    'fontWeight': '700'
  }
}
