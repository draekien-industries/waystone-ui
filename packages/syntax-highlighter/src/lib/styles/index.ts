export * from './a11yDark.generated';
export * from './a11yLight.generated';
export * from './agate.generated';
export * from './anOldHope.generated';
export * from './androidstudio.generated';
export * from './arduinoLight.generated';
export * from './arta.generated';
export * from './ascetic.generated';
export * from './atomOneDarkReasonable.generated';
export * from './atomOneDark.generated';
export * from './atomOneLight.generated';
export * from './brownPaper.generated';
export * from './codepenEmbed.generated';
export * from './colorBrewer.generated';
export * from './dark.generated';
export * from './defaultStyle.generated';
export * from './devibeans.generated';
export * from './docco.generated';
export * from './far.generated';
export * from './felipec.generated';
export * from './foundation.generated';
export * from './githubDarkDimmed.generated';
export * from './githubDark.generated';
export * from './github.generated';
export * from './gml.generated';
export * from './googlecode.generated';
export * from './gradientDark.generated';
export * from './gradientLight.generated';
export * from './grayscale.generated';
export * from './hybrid.generated';
export * from './idea.generated';
export * from './intellijLight.generated';
export * from './irBlack.generated';
export * from './isblEditorDark.generated';
export * from './isblEditorLight.generated';
export * from './kimbieDark.generated';
export * from './kimbieLight.generated';
export * from './lightfair.generated';
export * from './lioshi.generated';
export * from './magula.generated';
export * from './monoBlue.generated';
export * from './monokaiSublime.generated';
export * from './monokai.generated';
export * from './nightOwl.generated';
export * from './nnfxDark.generated';
export * from './nnfxLight.generated';
export * from './nord.generated';
export * from './obsidian.generated';
export * from './pandaSyntaxDark.generated';
export * from './pandaSyntaxLight.generated';
export * from './paraisoDark.generated';
export * from './paraisoLight.generated';
export * from './pojoaque.generated';
export * from './purebasic.generated';
export * from './qtcreatorDark.generated';
export * from './qtcreatorLight.generated';
export * from './rainbow.generated';
export * from './routeros.generated';
export * from './schoolBook.generated';
export * from './shadesOfPurple.generated';
export * from './srcery.generated';
export * from './stackoverflowDark.generated';
export * from './stackoverflowLight.generated';
export * from './sunburst.generated';
export * from './tokyoNightDark.generated';
export * from './tokyoNightLight.generated';
export * from './tomorrowNightBlue.generated';
export * from './tomorrowNightBright.generated';
export * from './vs.generated';
export * from './vs2015.generated';
export * from './xcode.generated';
export * from './xt256.generated';
