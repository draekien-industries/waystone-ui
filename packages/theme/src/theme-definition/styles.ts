export const styles = {
  shadows: {
    none: 'none',
    small:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    medium:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 2px hsla(0, 0%, 0%, 0.05), 0 0.6px 0.8px hsla(0, 0%, 0%, 0.02), 0 1.8px 3px hsla(0, 0%, 0%, 0.045), 0 7px 12px hsla(0, 0%, 0%, 0.09)',
    large:
      'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 4px hsla(0, 0%, 0%, 0.05), 0 1.2px 1.6px hsla(0, 0%, 0%, 0.02), 0 3.6px 6px hsla(0, 0%, 0%, 0.045), 0 14px 24px hsla(0, 0%, 0%, 0.09)',
  },
  sizes: {
    'extra-small': '1em',
    small: '5em',
    medium: '10em',
    large: '15em',
    'extra-large': '20em',
    half: '50%',
    full: '100%',
  },
} as const;
