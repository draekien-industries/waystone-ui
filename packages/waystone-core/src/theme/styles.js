export const styles = {
    shadows: {
        sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
        md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
        lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
        'sm-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.125em 0.25em -0.125em #0b0b0bA0',
        'md-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.5em 1em -0.5em #0b0b0bA0',
        'lg-dark': '0 0 0.5em 0 #1b1b1bFF, 0 1em 2em -1em #0b0b0bA0',
    },
    sizes: {
        xs: '1em',
        sm: '5em',
        md: '10em',
        lg: '15em',
        xl: '20em',
        half: '50%',
        full: '100%',
    },
};
export default styles;
