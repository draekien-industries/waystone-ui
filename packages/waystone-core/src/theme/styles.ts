export const styles = {
  shadows: {
    sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
    md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
    lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
    'sm-dark':
      '0 0 0.5em 0 rgba(27, 27, 27, 1), 0 0.5 1em -0.5em rgba(11, 11, 11, 0.73)',
    'md-dark':
      '0 0 1em 0 rgba(27, 27, 27, 1), 0 1em 2em -1em rgba(11, 11, 11, 0.73)',
    'lg-dark':
      '0 0 2em 0 rgba(27, 27, 27, 1), 0 2em 4em -2em rgba(11, 11, 11, 0.73)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
};

export default styles;
