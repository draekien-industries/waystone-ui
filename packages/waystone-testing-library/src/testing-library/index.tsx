'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
