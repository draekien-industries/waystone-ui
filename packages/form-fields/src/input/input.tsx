import type { FullWidthAttributes, WidthAttributes } from '@waystone/types';
import {
  forwardRef,
  type ComponentPropsWithoutRef,
  type ForwardedRef,
} from 'react';
import { inputCss } from './input.styles';

export type InputProps = {
  /** The ID of the input */
  id: string;
} & ComponentPropsWithoutRef<'input'> &
  FullWidthAttributes &
  WidthAttributes;

const InputWithRef = (
  { fullWidth, width, minWidth, maxWidth, ...rest }: InputProps,
  ref: ForwardedRef<HTMLInputElement>
) => (
  <input
    ref={ref}
    {...rest}
    sx={inputCss({ fullWidth, width, minWidth, maxWidth })}
  />
);

export const Input = forwardRef(InputWithRef);
