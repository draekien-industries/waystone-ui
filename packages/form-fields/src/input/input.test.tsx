import { render } from '@waystone/testing-library';
import { Input } from './input';
import { inputCss } from './input.styles';

describe('<Input />', () => {
  describe('component', () => {
    it('should match snapshot', () => {
      const { container } = render(<Input id="test-input" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('css', () => {
    it('should match snapshot', () => {
      const css = inputCss({});

      expect(css).toMatchSnapshot();
    });

    it('should have 100% when fullWidth is true', () => {
      const css = inputCss({ fullWidth: true });

      expect(css.width).toBe('100%');
    });

    it('should use provided width when fullWidth is undefined', () => {
      const css = inputCss({ width: '1rem' });

      expect(css.width).toBe('1rem');
    });

    it('should override wdith with 100% when fullWidth is true', () => {
      const css = inputCss({ width: '1rem', fullWidth: true });

      expect(css.width).toBe('100%');
    });

    it('should use minWidth and maxWidth', () => {
      const css = inputCss({ minWidth: '1rem', maxWidth: '2rem' });

      expect(css.minWidth).toBe('1rem');
      expect(css.maxWidth).toBe('2rem');
    });
  });
});
