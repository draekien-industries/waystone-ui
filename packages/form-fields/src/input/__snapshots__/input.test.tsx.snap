// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component should match snapshot 1`] = `
<div>
  <input
    class="css-1g6xmhl-InputWithRef"
    id="test-input"
  />
</div>
`;

exports[`<Input /> css should match snapshot 1`] = `
{
  ":enabled:active": {
    "outlineColor": [Function],
  },
  ":enabled:not(active):focus-visible": {
    "outlineColor": [Function],
  },
  "backgroundColor": [Function],
  "border": "1px solid",
  "borderColor": "b-400",
  "borderRadius": "small",
  "color": "text",
  "maxWidth": undefined,
  "minWidth": undefined,
  "outlineColor": [Function],
  "outlineOffset": 2,
  "outlineStyle": "solid",
  "outlineWidth": 4,
  "padding": "extra-small",
  "variant": "text.body",
  "width": undefined,
}
`;
